examples/src/main/java/com/expediagroup/sdk/rapid/examples/scenarios/booking/MultiRoomHoldAndResumeBookScenario.java

```diff
@@ -0,0 +1,88 @@
+         * Up to 8 rooms may be requested or booked at once.
+         */
+        logger.info("Getting property availability for test property: [{}]", Constants.TEST_PROPERTY_ID);
+        List<String> occupancy = Arrays.asList("2", "2");
+        ShopService shopService = new ShopService();
+        List<Property> propertyAvailabilityList = shopService.getMultipleRoomsPropertiesAvailability(this.rapidPartnerSalesProfile, occupancy);
+
+        if (propertyAvailabilityList != null && !propertyAvailabilityList.isEmpty()) {
+            logger.info("Property Availability found for property id: [{}] with status: [{}]",
+                    propertyAvailabilityList.get(0).getPropertyId(),
+                    propertyAvailabilityList.get(0).getStatus());
+        } else {
+            logger.error("No property availability found for the test property.");
+            return;
+        }
+
+        // Check room prices for the property
+        logger.info("Checking room prices for property id: [{}]...", Constants.TEST_PROPERTY_ID);
+        PropertyAvailability propertyAvailability = (PropertyAvailability) propertyAvailabilityList.get(0);
+        RoomPriceCheck roomPriceCheck = null;
+
+        roomPriceCheck = shopService.checkRoomPrices(propertyAvailability);
+        logger.info("Room price check status: [{}]", roomPriceCheck.getStatus());
+
+        /*
+         * Book multiple rooms in the property.
+         * The Booking link from your previous Price Check response expires after a short period.
+         * If you receive an HTTP 503 error upon your first attempt, the link has likely expired.
+         * Obtain a new link and attempt your booking again.
+         */
+        logger.info("Booking 2 rooms with [hold=true] in test property: [{}]...", Constants.TEST_PROPERTY_ID);
+        BookService bookService = new BookService();
+        ItineraryCreation itineraryCreation = bookService.createBookingWithHold(roomPriceCheck, occupancy);
+        logger.info("Booking with hold Success. Itinerary id: [{}]. Link to resume booking: [{}]",
+                itineraryCreation.getItineraryId(), itineraryCreation.getLinks().getResume().getHref());
+
+        // Resume the booking process
+        logger.info("Resuming the booking process...");
+        Response<Nothing> response = bookService.resumeBooking(itineraryCreation);
+        logger.info("Resume booking response status: [{}]", response.getStatusCode());
+
+        // Make a retrieve call to verify the booking has been resumed properly.
+        logger.info("Getting itinerary by itinerary id: [{}] to verify the booking has been resumed successfully...",
+                itineraryCreation.getItineraryId());
+        Itinerary itinerary = bookService.getReservationByItineraryId(itineraryCreation);
+
+        logger.info("Itinerary rooms status after resume booking:");
+        itinerary.getRooms().forEach(room ->
```

Comment:
Let's `assert itinerary.getRooms() != null;` before this.

Reply:
We've discussed this in the previous scenario, and Ibrahim asked that we don't add assertions and null checks to keep the code clean and focused on the business scenario.